Temperature monitor 2.0

THX723 opened the pull request:
Made temperature monitoring & logging optional, and with configurable settings.

Temperature threshold logging for Warning, and Caution:
```
export TEMPERATURE_CAUTION=50000
export TEMPERATURE_WARNING=68000
```

Fixed-interval temperature logging (instantaneous and max):
```
export TEMPERATURE_INTERVAL=60
```

Post-archiving/music sync temperature logging:
```
export TEMPERATURE_POSTARCHIVE=true
```

THX723 committed:
Update teslausb_setup_variables.conf.sample

Add new section for Temperature Monitor configuration.
The new parameters are:
export TEMPERATURE_CAUTION=50000
export TEMPERATURE_WARNING=68000
export TEMPERATURE_INTERVAL=60
export TEMPERATURE_POSTARCHIVE=true

THX723 committed:
Create temperature_monitor

New file for Temperature Monitor implements.

THX723 committed:
Update archiveloop

Removed previous log_coretemp function. Replaced with call to new temperature_monitor.

THX723 committed:
Update teslausb_setup_variables.conf.sample

fixed spellings

THX723 committed:
Update temperature_monitor

THX723 committed:
Update teslausb_setup_variables.conf.sample

minor rephrasing.

THX723 committed:
Update archiveloop

fix typo

stauffenberg2020 commented:
Can the user get this information (caution, warning, post-archive temperature if archive is complete) through existing notification mechanism?

THX723 commented:
> Can the user get this information (caution, warning, post-archive temperature if archive is complete) through existing notification mechanism?

I have thought about it. Just getting the base code out of the way first. Would be fairly simple to add using `send-push-message`. Obviously will only work if there's an internet connection. For most, that means during archiving at home.

THX723 committed:
Update temperature_monitor

added push notifications for caution, warning, and post-archive cases.

THX723 committed:
Update temperature_monitor

fixed typos and tab spacings

THX723 committed:
Update temperature_monitor

THX723 committed:
Update teslausb_setup_variables.conf.sample

Revised verbiage and added statement for push notification.

THX723 committed:
Update temperature_monitor

THX723 committed:
Update temperature_monitor

Remove logging messages, if already push notifying.

THX723 committed:
Update teslausb_setup_variables.conf.sample

THX723 committed:
Update teslausb_setup_variables.conf.sample

THX723 committed:
Refactor keep car awake

Keep Car Away 2.0
-BLE/Tessie without using Sentry Mode.
-TeslaFi error handler
-install/retain 'bluez' 'pi-bluetooth' for BLE

THX723 committed:
Merge branch 'main-dev' into Temperature-Monitor-2.0

THX723 commented:
I accidentally merged the wrong things together. Will have to delete this PR and resubmit.

THX723 commented:
Superseded by [PR #982](https://github.com/marcone/teslausb/pull/982).
